src/config/settings.py

```
@@ -330,18 +315,212 @@ def load(
+        if env is None:
+            from fancyrag.utils.env import load_project_dotenv
+
+            load_project_dotenv()
+            source: Mapping[str, str] = os.environ
+        else:
+            source = env
+
+        def _require(key: str) -> str:
+            value = (source.get(key) or "").strip()
+            if not value:
+                raise ValueError(f"Missing required environment variable: {key}")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)